### PR TITLE
chore: polish pull request template

### PR DESCRIPTION
**Pull request by Wendong-Fan**
💡 **User Story:**
As a developer,
I want to change the PULL_REQUEST_TEMPLATE
So that all developers can use the new template

🛒 **Types of Changes:**
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)
- [ ] Test (only for test)

🚀 **Acceptance Criteria:**
[Clearly define the conditions that must be met for the story to be considered complete]
[List any specific requirements, features, or behaviors the new item should exhibit]

🔀 **Related User Stories or Open Issue:**
- related to #278 
- related to #294

🛠️ **Technical Details:**
/

📑 **Tasks:**
/

🔑 **Dependencies and Pre-requisites:**
/

🤝 **Required Prerequisites:**
- [X] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+[List specific tasks or steps required to implement the user story]
+[Break down the development process into manageable tasks]
+...
 
-Go over all the following points, and put an `x` in all the boxes that apply.
-If you are unsure about any of these, don't hesitate to ask. We are here to help!
+🔑 **Dependencies:**
+[Specify any dependencies that need to be resolved before working on this story]
 
-- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
```
Comment (Benjamin-eecs) [inline .github/PULL_REQUEST_TEMPLATE.md:34]: Please keep this entry to remind users to read `CONTRIBUTING.md`

Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md:34]: thank you for your suggestion, have changed Required Prerequisites part

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
+As a developer,
+I want to [describe the action or functionality you want to add]
+So that [explain the benefit or value this new functionality will bring]
 
+🛒 **Types of Changes:**
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds core functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)
 - [ ] Documentation (update in the documentation)
 - [ ] Example (update in the folder of example)
```
Comment (dandansamax) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Here we can add a "Test" type for test only modifications.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,37 +1,38 @@
...
 
-## Checklist
+📑 **Tasks:**
+[List specific tasks or steps required to implement the user story]
+[Break down the development process into manageable tasks]
+...
 
-Go over all the following points, and put an `x` in all the boxes that apply.
-If you are unsure about any of these, don't hesitate to ask. We are here to help!
+🔑 **Dependencies:**
```
Comment (dandansamax) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: This is nice, but may be ambiguous with "package dependencies"?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,37 +1,38 @@
-## Description
-
-Describe your changes in detail.
-
-## Motivation and Context
-
-Why is this change required? What problem does it solve?
-If it fixes an open issue, please link to the issue here.
-You can use the syntax `close #15213` if this solves the issue #15213
```
Comment (dandansamax) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Maybe it's better to keep "Related issue".

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,37 +1,38 @@
...
+
+🔀 **Related User Stories:**
+- [List related user stories that contribute to this story]
+- [User Story 1]
+- [User Story 2]
 
-- [ ] Subtask 1
-- [ ] Subtask 2
-- [ ] Subtask 3
+🛠️ **Technical Details:**
```
Comment (dandansamax) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: This is very nice. How about add "third-party packages" in the description.

**Review by dandansamax — CHANGES_REQUESTED** [review]
Thank you for your work! Here are some of my suggestions:

1. This template is more likely for issue of a "feature request", since "user story", "tasks", and "dependencies" should be decided and disscussed before we implement it. When a pull request is opened, we won't discuss whether we need  this feature or not, but only care about the code implementation.
2. Terms like "Acceptance Criteria" and "Related User Stories" is hard to understand. Do we really need them? In my opinion, developer will simply ignore them if they cannot easily get the idea in 10 seconds.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: Added

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md]: Adjusted

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Added

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: Added

**Comment by Wendong-Fan** [comment]
Hi @dandansamax , I noticed #278 is related to this user story, so I also added you as a contributor. If it's necessary we can take a meeting to discuss about the template together.

**Comment by dandansamax** [comment]
> Hi @dandansamax , I noticed #278 is related to this user story, so I also added you as a contributor. If it's necessary we can take a meeting to discuss about the template together.

@Wendong-Fan I'm happy to discuss it. Could you host a meeting and let it open to all developers in the slack? This template is highly related to our workflow and should be confirmed by all members.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,37 +1,40 @@
...
-
-What types of changes does your code introduce? Put an `x` in all the boxes that apply:
+💡 **User Story:**
+As a developer,
+I want to [describe the action or functionality you want to add]
+So that [explain the benefit or value this new functionality will bring]
 
+🛒 **Types of Changes:**
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds core functionality)
```
Comment (Obs01ete) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Let's add the following after New feature:
Improvement (non-breaking change which improves existing core functionality)

**Review by Obs01ete — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (Wendong-Fan) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Hi Dmitrii, thank you for your suggestion and sorry for the late reply, have added it

**Comment by Wendong-Fan** [comment]
After this item is closed will open another PR to polish the issue template

**Review by Benjamin-eecs — APPROVED** [review]
LGTM! cc @Obs01ete

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,37 +1,39 @@
...
-- [ ] Breaking change (fix or feature that would cause existing functionality to change)
```
Comment (Obs01ete) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Why is the Breaking change gone? Revert.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,37 +1,39 @@
-## Description
-
-Describe your changes in detail.
-
-## Motivation and Context
-
-Why is this change required? What problem does it solve?
```
Comment (Obs01ete) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: I like the original Description and Motivation and Context chapters. Why changing them?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -1,37 +1,39 @@
...
+So that [explain the benefit or value this new functionality will bring]
 
+🛒 **Types of Changes:**
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds core functionality)
-- [ ] Breaking change (fix or feature that would cause existing functionality to change)
+- [ ] Improvement (non-breaking change which improves existing core functionality)
 - [ ] Documentation (update in the documentation)
 - [ ] Example (update in the folder of example)
+- [ ] Test (only for test)
```
Comment (Obs01ete) [inline .github/PULL_REQUEST_TEMPLATE.md:12]:
```suggestion
- [ ] Test (only tests are modified or added)
```

**Review by Obs01ete — CHANGES_REQUESTED** [review]
In general I go not see the need for this PR altogether. I find the original template better.

**Comment by lightaime** [comment]
Hi @Wendong-Fan. I share a similar concern with @dandansamax. `User story` seems to be a term that is related to agile. It may be hard for developers who are not familiar with agile to understand it. How about we keep it simple as something like `Pull Request Description`?

**Comment by Wendong-Fan** [comment]
Hi @lightaime  @Obs01ete @dandansamax @Benjamin-eecs ,

I apologize for the delayed response, as I was away on vacation. I'm not an agile coach or scrum master, my intention was simply to introduce beneficial practices from my work experience to enhance our team's workflow.

The core principle behind these proposed changes was to customize our processes to better suit our specific needs and foster improvements. However, after further consideration and taking into account your feedback, I realize that these changes may not be delivering the expected value and could even cause confusion.

In light of this, I propose that we revert to our previous, simpler template. Consequently, I will go ahead and close this pull request. I appreciate your understanding and cooperation.

**Pull request closed without merging**